src/main/java/org/tikv/common/region/RegionManager.java

```diff
@@ -191,17 +192,23 @@ public Pair<TiRegion, TiStore> getRegionStorePairByKey(
       Peer peer = region.getCurrentReplica();
       store = getStoreById(peer.getStoreId(), backOffer);
     } else {
-      outerLoop:
+      List<TiStore> tiflashStores = new ArrayList<>();
       for (Peer peer : region.getLearnerList()) {
         TiStore s = getStoreById(peer.getStoreId(), backOffer);
         for (Metapb.StoreLabel label : s.getStore().getLabelsList()) {
           if (label.getKey().equals(storeType.getLabelKey())
               && label.getValue().equals(storeType.getLabelValue())) {
-            store = s;
-            break outerLoop;
+            tiflashStores.add(s);
           }
         }
       }
+      // select a tiflash with RR strategy
+      if (tiflashStores.size() > 0) {
+        store =
+            tiflashStores.get(TiFlashStoreIndex > tiflashStores.size() - 1 ? 0 : TiFlashStoreIndex);
+        TiFlashStoreIndex++;
```

Comment:
Considering the method `getRegionStorePairByKey` might call by multiple threads, we might change the idx member to be atomic.